src/MSBuild.Sdk.SqlProj.Aspire/DataTierApplicationBuilderExtensions.cs

```diff
@@ -0,0 +1,65 @@
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.Lifecycle;
+using Microsoft.Build.Locator;
+using MSBuild.Sdk.SqlProj.Aspire;
+
+namespace Aspire.Hosting;
+
+public static class DataTierApplicationBuilderExtensions
+{
+    /// <summary>
+    /// Adds a data-tier application resource to the application based on a referenced project.
+    /// </summary>
+    /// <typeparam name="TProject">Type that represents the project that produces the data-tier application package.</typeparam>
+    /// <param name="builder">An <see cref="IDistributedApplicationBuilder"/> instance to add the data-tier application to.</param>
+    /// <param name="name">Name of the resource.</param>
+    /// <returns>An <see cref="IResourceBuilder{T}"/> that can be used to further customize the resource.</returns>
+    public static IResourceBuilder<DataTierApplicationResource> AddDataTierApplication<TProject>(this IDistributedApplicationBuilder builder, string name)
+        where TProject : IProjectMetadata, new()
+    {
+        if (!MSBuildLocator.IsRegistered)
+        {
+            MSBuildLocator.RegisterInstance(MSBuildLocator.QueryVisualStudioInstances().OrderByDescending(
+                instance => instance.Version).First());
+        }
+
+        var resource = new DataTierApplicationResource(name);
+        
+        return builder.AddResource(resource)
+                      .WithAnnotation(new TProject());
+    }
+
+    /// <summary>
+    /// Adds a data-tier application resource to the application based on a data-tier application package file.
```

Comment:
Maybe: Adds a data-tier application resource to the application based on a data-tier application package file (.dacpac).